### ISSUE TITLE
Importing aws:elasticache/replicationGroup:ReplicationGroup without auth token yields a resource that Pulumi constantly wants to replace

### ISSUE THREAD

**Issue by simonwhitaker** [pulumi/pulumi-aws#4644]
### Describe what happened

I imported a Redis OSS replication group, on which no auth token has been set, using:

```
pulumi import aws:elasticache/replicationGroup:ReplicationGroup my_replication_group replication-group-id
```

That worked, and gave me the following code, which I added to my repo:

```
import * as pulumi from "@pulumi/pulumi";
import * as aws from "@pulumi/aws";

const my_replication_group = new aws.elasticache.ReplicationGroup("my_replication_group", {
    atRestEncryptionEnabled: true,
    autoMinorVersionUpgrade: true,
    automaticFailoverEnabled: true,
    clusterMode: "disabled",
    description: "Description here",
    engineVersion: "7.1",
    ipDiscovery: "ipv4",
    maintenanceWindow: "sun:03:00-sun:04:00",
    multiAzEnabled: true,
    networkType: "ipv4",
    nodeType: "cache.t4g.medium",
    numCacheClusters: 2,
    parameterGroupName: "default.redis7",
    port: 6379,
    replicationGroupId: "replication-group-id",
    securityGroupIds: ["sg-id-here"],
    snapshotRetentionLimit: 1,
    snapshotWindow: "00:00-01:00",
    subnetGroupName: "subnet-group-name-here",
}, {
    protect: true,
});
```

I copied and pasted the resource code into my codebase. Then I ran `pulumi up`, and it failed:

```
$ pulumi up
Previewing update (phoebeai/production)

View in Browser (Ctrl+O): https://app.pulumi.com/phoebeai/monorepo/production/previews/18e4a31c-cf93-45aa-87f2-452949176820

     Type                                 Name                       Plan        Info
     pulumi:pulumi:Stack                  monorepo-production                    1 error
 +-  └─ aws:elasticache:ReplicationGroup  my_replication_group  replace     [diff: +authTokenUpdateStrategy]; 1 error

Diagnostics:
  pulumi:pulumi:Stack (monorepo-production):
    error: preview failed

  aws:elasticache:ReplicationGroup (my_replication_group):
    error: unable to replace resource "urn:pulumi:production::monorepo::aws:elasticache/replicationGroup:ReplicationGroup::my_replication_group"
    as it is currently marked for protection. To unprotect the resource, remove the `protect` flag from the resource in your Pulumi program and run `pulumi up`

    [Pulumi Copilot] Would you like help with these diagnostics?
    https://app.pulumi.com/phoebeai/monorepo/production/previews/18e4a31c-cf93-45aa-87f2-452949176820?explainFailure
```

We don't have an auth token configured on this replication group, and `authTokenUpdateStrategy` is not defined in the resource definition (i.e. if I run `aws elasticache describe-replication-groups`).

I can't find any way to change this definition such that it can be cleanly applied without replacing my replication group, which obviously I don't want to do.

### Sample program

import * as pulumi from "@pulumi/pulumi";
import * as aws from "@pulumi/aws";

const my_replication_group = new aws.elasticache.ReplicationGroup("my_replication_group", {
    atRestEncryptionEnabled: true,
    autoMinorVersionUpgrade: true,
    automaticFailoverEnabled: true,
    clusterMode: "disabled",
    description: "Description here",
    engineVersion: "7.1",
    ipDiscovery: "ipv4",
    maintenanceWindow: "sun:03:00-sun:04:00",
    multiAzEnabled: true,
    networkType: "ipv4",
    nodeType: "cache.t4g.medium",
    numCacheClusters: 2,
    parameterGroupName: "default.redis7",
    port: 6379,
    replicationGroupId: "replication-group-id",
    securityGroupIds: ["sg-id-here"],
    snapshotRetentionLimit: 1,
    snapshotWindow: "00:00-01:00",
    subnetGroupName: "subnet-group-name-here",
}, {
    protect: true,
});

### Log output

Previewing update (phoebeai/production)

View in Browser (Ctrl+O): https://app.pulumi.com/phoebeai/monorepo/production/previews/18e4a31c-cf93-45aa-87f2-452949176820

     Type                                 Name                       Plan        Info
     pulumi:pulumi:Stack                  monorepo-production                    1 error
 +-  └─ aws:elasticache:ReplicationGroup  my_replication_group  replace     [diff: +authTokenUpdateStrategy]; 1 error

Diagnostics:
  pulumi:pulumi:Stack (monorepo-production):
    error: preview failed

  aws:elasticache:ReplicationGroup (my_replication_group):
    error: unable to replace resource "urn:pulumi:production::monorepo::aws:elasticache/replicationGroup:ReplicationGroup::my_replication_group"
    as it is currently marked for protection. To unprotect the resource, remove the `protect` flag from the resource in your Pulumi program and run `pulumi up`

    [Pulumi Copilot] Would you like help with these diagnostics?
    https://app.pulumi.com/phoebeai/monorepo/production/previews/18e4a31c-cf93-45aa-87f2-452949176820?explainFailure

### Affected Resource(s)

aws:elasticache/replicationGroup:ReplicationGroup

### Output of `pulumi about`

CLI
Version      3.136.1
Go Version   go1.23.2
Go Compiler  gc

Plugins
KIND      NAME    VERSION
resource  aws     6.53.0
resource  awsx    2.16.0
resource  docker  4.5.6
resource  docker  3.6.1
language  nodejs  unknown

Host
OS       darwin
Version  15.0.1
Arch     arm64

This project is written in nodejs: executable='/Users/simon/.local/state/fnm_multishells/72333_1729060485469/bin/node' version='v20.14.0'

Current Stack: phoebeai/monorepo/production

TYPE                                               URN
[Resources removed]

Found no pending operations associated with phoebeai/production

Backend
Name           pulumi.com
URL            https://app.pulumi.com/simonatphoebe
User           simonatphoebe
Organizations  simonatphoebe, phoebeai
Token type     personal

Dependencies:
NAME            VERSION
@pulumi/aws     6.53.0
@pulumi/awsx    2.16.0
@pulumi/pulumi  3.134.1
@types/node     18.19.53
typescript      5.6.2

Pulumi locates its logs in /var/folders/0w/bxd_wv6510bb2c2l1mfvhgz80000gn/T/ by default

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by t0yv0** [comment]
Thank you for reporting this issue! Looking at Terraform code this provider is linking, there are some possible values for authTokenUpdateStrategy.


```
const (
	AuthTokenUpdateStrategyTypeSet    AuthTokenUpdateStrategyType = "SET"
	AuthTokenUpdateStrategyTypeRotate AuthTokenUpdateStrategyType = "ROTATE"
	AuthTokenUpdateStrategyTypeDelete AuthTokenUpdateStrategyType = "DELETE"
)

```

The default value is AuthTokenUpdateStrategyTypeRotate.

```
I can't find any way to change this definition such that it can be cleanly applied without replacing my replication group, which obviously I don't want to do.
```

It seems that possibly editing the authTokenUpdateStrategy and setting it to one of the above values should allow `pulumi up` to succeed without replacing. Did you try `authoTokenUpdateStrategy="ROTATE"`?

If you could paste the `pulumi stack export` results pertinent to this aws.elasticache.ReplicationGroup resource in your sample program we could reproduce on our end against the state you are getting.

**Comment by simonwhitaker** [comment]
Yes, I tried adding `authoTokenUpdateStrategy="ROTATE"`. (And indeed, all the other valid values for that field, and empty string, and null, and undefined.) None of those approaches worked.

I ended up creating a new replication group with Pulumi, switching my traffic over to that, then removing the old replication group. 😅 So I won't get anything useful in `pulumi stack export` I'm afraid.
